packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts

```diff
@@ -15,3 +18,165 @@ export enum Permissions {
+    });
+  }
+
+  getVersion(version: number): LayerVersionMetadata {
+    return this.versionMap.get(version);
+  }
+
+  listVersions(): number[] {
+    return Array.from(this.versionMap.keys());
+  }
+  getLatestVersion(): number {
+    return Array.from(this.versionMap.keys()).reduce((a, b) => Math.max(a, b));
+  }
+}
+
+class LayerVersionState implements LayerVersionMetadata {
+  permissions: LayerPermission[];
+  constructor(value) {
+    this.permissions = [];
+    value.forEach(permission => {
+      if (permission.type === Permissions.public) {
+        const permissionPublic: PublicLayer = {
+          type: Permissions.public,
+        };
+        this.permissions.push(permissionPublic);
+      }
+      if (permission.type === Permissions.private) {
+        const permissionPrivate: PrivateLayer = {
+          type: Permissions.private,
+        };
+
+        this.permissions.push(permissionPrivate);
+      }
+      if (permission.type === Permissions.awsOrg) {
+        const orgsPermissions: OrgsLayer = {
+          type: Permissions.awsOrg,
+          orgs: permission.orgs,
+        };
+        this.permissions.push(orgsPermissions);
+      }
+      if (permission.type === Permissions.awsAccounts) {
+        const accountsPermissions: AccountsLayer = {
+          type: Permissions.awsAccounts,
+          accounts: permission.accounts,
+        };
+        this.permissions.push(accountsPermissions);
+      }
+    });
+  }
+
+  listAccoutAccess(): string[] {
+    let accounts: string[] = [];
+    if (!(this.permissions === undefined || this.permissions.length == 0)) {
+      this.permissions.forEach(permission => {
+        if (permission.type === Permissions.awsAccounts) {
+          accounts = permission.accounts;
+        }
+      });
+    }
+    return accounts;
+  }
+
+  listOrgAccess(): string[] {
+    let orgs: string[] = [];
+    if (!(this.permissions === undefined || this.permissions.length == 0)) {
+      this.permissions.forEach(permission => {
+        if (permission.type === Permissions.awsOrg) {
+          orgs = permission.orgs;
+        }
+      });
+    }
+    return orgs;
+  }
+
+  isPrivate(): boolean {
+    let retVal: boolean = false;
+    if (!(this.permissions === undefined || this.permissions.length == 0)) {
+      this.permissions.forEach(permission => {
+        if (permission.type === Permissions.private) {
+          retVal = true;
+        }
+      });
+    }
+    return retVal;
+  }
+
+  isPublic(): boolean {
+    let retVal: boolean = false;
+    if (!(this.permissions === undefined || this.permissions.length == 0)) {
+      this.permissions.forEach(permission => {
+        if (permission.type === Permissions.public) {
+          retVal = true;
+        }
+      });
+    }
+    return retVal;
+  }
+}
+
+export const layerMetadataFactory: LayerMetadataFactory = (context: any, layerName: string, isPush: boolean = false): LayerMetadata => {
+  let projectBackendDirPath;
+  if (!isPush) {
```

Comment:
better to use positive conditional whenever possible. So revers this to be `if (isPush)`

Better yet, why not have this method take in the backendDirPath directly instead of the isPush flag. That way no if statement is required